### ISSUE TITLE
Cloudflare security Captchas or JavaScript browser checks

### ISSUE THREAD

**Issue by davidfischer** [readthedocs/readthedocs.org#4509]
Documentation sites (`*.readthedocs.io` and others) are cached for us by Cloudflare who also does SSL termination on those domains. Cloudflare can block certain requests or issue challenges which take the form of a captcha or a javascript check ("checking your browser before accessing...") when the requestor seems suspicious. Suspicion is based on a number of factors including details of the request (requesting `/wp-login.php` for example) or previous requests Cloudflare has seen from the same IP.

In this issue, my goal is to collect any data people are willing to share on challenges or blocks from Cloudflare that were received while accessing Read the Docs. Ideally, users should not get challenged or blocked or this should be extremely infrequent and/or justified.

Our security settings are tuned down a bit and generally I haven't seen any legitimate requests get blocked. However, if users are experiencing problems, I want to hear about it. We don't have any problems with users connecting via Tor and in testing our Cloudflare setup did not challenge requests coming from the Tor Browser. However, if Tor users or users in specific countries are having problems, please share and we will fix it.

* [Cloudflare's Tor guidance](https://support.cloudflare.com/hc/en-us/articles/203306930-Does-Cloudflare-block-Tor-)
* [Tor appeal](https://gitlab.com/iblech/tor-appeal/)

### Captcha
<img width="715" alt="screen shot 2018-08-13 at 9 34 32 am" src="https://user-images.githubusercontent.com/185043/44045036-413d323a-9edc-11e8-8b6d-1248b574364a.png">

### Javascript challenge
![tor-visitor-javascript-challenge](https://user-images.githubusercontent.com/185043/44045348-1c0c30b4-9edd-11e8-9209-796836494d58.png)

**Comment by humitos** [comment]
We got our first report from a Tor user: https://github.com/readthedocs/readthedocs.org/issues/7598

@davidfischer since you created this issue, I'm pinging you because I suppose you may be interested to take a look.

**Comment by Terrance** [comment]
I've just opened #8251, which looks to be relevant here.

**Comment by Idlusen** [comment]
Using Tor I consistently am presented with the captcha page.
As a workaround I can use a webproxy but this is tedious.

**Comment by davidfischer** [comment]
@Idlusen Does this occur on all RTD documentation? For example, does this also occur on https://docs.readthedocs.io? Can you describe your Tor setup a bit? This does not happen for me at least with the Tor browser from torproject.org although it could be based on your exit node. The way these reputation challenges are supposed to work (and using Tor doesn't strictly mean you'll get a challenge) is that you get a challenge but after doing the captcha, you shouldn't get challenged again during the session. Does that work correctly for you?

**Comment by Idlusen** [comment]
It occurred on all the subdomains I visited, namely trio, arrow, asks and beets last week but to my surprise it doesn’t today on all these plus docs. Maybe I was very unlucky with my exit nodes…
I am using the Tor browser fetched by the Arch Linux package.
I am reluctant to complete captchas as they can fail several times for no apparent reason so I switched immediately to the webproxy but I’ll try next time I get challenged.
